msticpy/common/keyvault_client.py

```diff
@@ -0,0 +1,1000 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Keyvault client - adapted from Bluehound code."""
+import base64
+from datetime import datetime
+import json
+from typing import List, Any, Optional
+import warnings
+
+from adal import AuthenticationContext, AdalError
+from azure.core.exceptions import ResourceNotFoundError
+from azure.keyvault.secrets import SecretClient, KeyVaultSecret
+from azure.identity import DeviceCodeCredential, InteractiveBrowserCredential
+from azure.mgmt.keyvault import KeyVaultManagementClient
+from azure.mgmt.keyvault.models import (
+    AccessPolicyEntry,
+    VaultProperties,
+    Sku,
+    KeyPermissions,
+    SecretPermissions,
+    CertificatePermissions,
+    Permissions,
+    VaultCreateOrUpdateParameters,
+    Vault,
+)
+from IPython.display import display, HTML
+import keyring
+from keyring.errors import KeyringError
+from msrest.authentication import BasicTokenAuthentication
+from msrestazure.azure_exceptions import CloudError
+import pandas.io.clipboard as pyperclip
```

Comment:
Just interested why you aren't using pyperclip natively?

Reply:
I just thought it was one less package to have to install.